src/main/scala/functions/RecursiveFunction.scala

```diff
@@ -0,0 +1,39 @@
+package com.knoldus
+package functions
+
+import scala.annotation.tailrec
+
+object RecursiveFunction extends App {
+
+  // ---------------------------------------------------------------------------------
+  def factorial(number: Int): BigInt = {
+    @tailrec
+    def factorialHelper(number: Int, accumulator: BigInt = 1): BigInt = {
+      if (number == 1) accumulator
+      else factorialHelper(number - 1 , number * accumulator)
+    }
+    factorialHelper(number)
+  }
+
```

Comment:
If the input is negative or zero, this method will enter an infinite loop. Always validate the input before performing any computation.